Packs/FeedUnit42/Integrations/FeedUnit42/FeedUnit42.py

```diff
@@ -0,0 +1,234 @@
+                        "value": indicator_object.get('name'),
+                        "type": UNIT42_TYPES_TO_DEMISTO_TYPES[key],
+                        "rawJSON": indicator_object,
+                        "fields": {
+                            "firstseenbysource": indicator_object.get('created'),
+                            "indicatoridentification": indicator_object.get('id'),
+                            "tags": list((set(indicator_object.get('labels'))).union(set(feed_tags))),
+                            "modified": indicator_object.get('modified'),
+                            "reportedby": 'Unit42',
+                        }
+                    })
+
+    return indicators
+
+
+def parse_relationships(indicators: list, relationships: list = [], pivots: list = []) -> list:
+    """Parse the relationships between indicators to attack-patterns, malware and campaigns.
+    Args:
+      indicators: a list of indicators.
+      relationships: a list of relationships.
+      pivots: a list of attack-patterns, malware and campaigns.
+    Returns:
+        A list of indicators, containing the indicators and the relationships between them.
+    """
+    for indicator in indicators:
+        indicator_id = indicator.get('fields', {}).get('indicatoridentification', '')
+        for relationship in relationships:
+            reference = ''
+            if indicator_id == relationship.get('source_ref'):
+                reference = relationship.get('target_ref', '')
+
+            elif indicator_id == relationship.get('target_ref'):
+                reference = relationship.get('source_ref', '')
+
+            field = ''
+            if reference:  # if there is a reference, get the relevant pivot
+                if reference.startswith('attack-pattern'):
+                    field = 'external_references'
+                    field_type = 'mitreexternalreferences'
+                elif reference.startswith('campaign'):
+                    field = 'name'
+                    field_type = 'campaign'
+                elif reference.startswith('malware'):
+                    field = 'name'
+                    field_type = 'malwarefamily'
+
+            if field:  # if there is a pivot, map the relevant data accordingly
+                for pivot in pivots:
+                    if pivot.get('id') == reference:
+                        pivot_field = pivot.get(field)
+                        if isinstance(pivot_field, str):
+                            # multiple malware or campaign names can be associated to an indicator
+                            if field_type in indicator.get('fields'):
+                                indicator['fields'][field_type].extend([pivot_field])
+                            else:
+                                indicator['fields'][field_type] = [pivot_field]
+                        else:  # a MITRE external reference
+                            indicator['fields'][field_type] = pivot_field
```

Comment:
shouldn't you have extend here in case `field_type in indicator.get('fields')`?

Reply:
i think not, as a mitre reference is a list of all references, while as campaign/malware name is just 1 out of a few possabilities.